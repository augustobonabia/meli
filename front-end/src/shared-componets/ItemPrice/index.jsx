import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

function Item(props) {
  const { price } = props;

  const renderCents = (decimals) => {
    if (decimals) {
      return <span className="cents">{decimals}</span>;
    }

    return null;
  };

  return (
    <span alt="Precio" className="price">
      <span alt="Moneda" className="currency">{price.currency}</span>
      <span alt="Monto" className="amount">{price.amount}</span>
      {renderCents(price.decimals)}
    </span>
  );
}

const pricePropType = PropTypes.shape({
  amount: PropTypes.number.isRequired,
  decimals: PropTypes.number.isRequired,
  currency: PropTypes.string.isRequired,
});

Item.propTypes = {
  price: pricePropType.isRequired,
};

export default Item;
export { pricePropType };
