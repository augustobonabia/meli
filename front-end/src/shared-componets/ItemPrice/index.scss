span.price {
  span.cents {
    margin-left: 0.2rem;
    font-size: 0.8rem;
    vertical-align: super;
  }
}