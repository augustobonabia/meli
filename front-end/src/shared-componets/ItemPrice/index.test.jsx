import React from 'react';
import ItemPrice from './index';
import renderer from 'react-test-renderer';

it('El precio se renderiza correctamente', () => {
  const price = {
    amount: 4850,
    decimals: 50,
    currency: '$',
  };

  const tree = renderer
    .create(<ItemPrice price={price} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
