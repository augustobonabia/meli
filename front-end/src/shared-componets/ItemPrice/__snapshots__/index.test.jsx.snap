// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El precio se renderiza correctamente 1`] = `
<span
  alt="Precio"
  className="price"
>
  <span
    alt="Moneda"
    className="currency"
  >
    $
  </span>
  <span
    alt="Monto"
    className="amount"
  >
    4.850
  </span>
  <span
    className="cents"
  >
    50
  </span>
</span>
`;
